src/utils/check-pr/check-pr.py

```diff
@@ -210,10 +212,53 @@ def deploy(self, filename: str) -> None:
             print(msg)
             subprocess.check_call(cmd, shell=True)
 
+    def register(self) -> None:
```

Comment:
i would simplify this.
There is no need to recreate the venv here 
you can reduce this whole function to the call to `registration.py create_cli_registration` + parsing the result. 
And call that function from deploy()

Reply:
also only call it if `--unattended` is specified see other comment

Reply:
Are you suggesting I use the same virtual environment as the one from the deploy stage? 
Otherwise I'm not sure how to access pip and python.

Reply:
yes that is what i am suggesting, No need to create a new one just for the registration.

Reply:
Done. Please review when able.